Source/NSubstitute/Routing/RouteFactory.cs

```diff
@@ -67,8 +67,8 @@ public IRoute RecordCallSpecification(ISubstituteState state)
         public IRoute RecordReplay(ISubstituteState state)
         {
             return new Route(RouteType.RecordReplay, new ICallHandler[] {
-                new ClearUnusedCallSpecHandler(state)
```

Comment:
I would keep `ClearUnusedCallSpecHandler` to explicitly clear the specs in case the implementations of `TrackLastCallHandler` changes.